nexus_constructor/pixel_options.py

```diff
@@ -62,12 +88,142 @@ def setupUi(self, parent_widget):
+            self.update_pixel_layout_visibility(True, False)
+            self._fill_single_pixel_fields(component_to_edit)
+
+        elif component_to_edit.get_field("detector_number") is not None:
+            self.entire_shape_radio_button.setChecked(True)
+            self.update_pixel_layout_visibility(False, True)
+            self._fill_entire_shape_fields(component_to_edit)
+
+        else:
+            self.no_pixels_button.setChecked(True)
+            self.pixel_options_stack.setVisible(False)
+
+    def _fill_single_pixel_fields(self, component_to_edit: Component):
+
+        x_pixel_offset = component_to_edit.get_field("x_pixel_offset")
+        y_pixel_offset = component_to_edit.get_field("y_pixel_offset")
+        detector_numbers = component_to_edit.get_field("detector_number")
+
+        if check_data_is_an_array(x_pixel_offset):
+
+            # If the pixel offset information is multidimensional then extra steps need to be taken to restore the info
+            n_rows, row_height = self._get_row_information(y_pixel_offset)
+            self.row_count_spin_box.setValue(n_rows)
+            self.row_height_spin_box.setValue(row_height)
+
+            n_cols, col_width = self._get_column_information(x_pixel_offset)
+            self.column_count_spin_box.setValue(n_cols)
+            self.column_width_spin_box.setValue(col_width)
+
+            first_id, start_counting_text, count_along_text = self._get_detector_number_information(
+                detector_numbers
+            )
+
+            self.first_id_spin_box.setValue(first_id)
+            self.start_counting_combo_box.setCurrentText(start_counting_text)
+            self.count_first_combo_box.setCurrentText(count_along_text)
+
+        else:
+
+            # If the pixel offset information represents a single pixel
+            self.first_id_spin_box.setValue(detector_numbers)
+
+    @staticmethod
+    def _get_row_information(y_pixel_offset: np.ndarray):
+
+        n_rows = y_pixel_offset.shape[0]
+        row_height = np.abs(y_pixel_offset[0][0] - y_pixel_offset[1][0])
+
+        return n_rows, row_height
+
+    @staticmethod
+    def _get_column_information(x_pixel_offset: np.ndarray):
+
+        n_cols = x_pixel_offset.shape[1]
+        col_width = np.abs(x_pixel_offset[0][1] - x_pixel_offset[0][0])
+
+        return n_cols, col_width
+
+    @staticmethod
+    def _get_detector_number_information(detector_numbers: np.ndarray):
+
+        first_id = np.amin(detector_numbers)
+
+        first_id_index = np.where(detector_numbers == first_id)
+        first_id_index = (first_id_index[0][0], first_id_index[1][0])
+
+        first_id_plus_one = first_id + 1
+
+        count_along_text = COLUMNS_TEXT
+        right_of_first_id = (first_id_index[0], first_id_index[1] + 1)
+        left_of_first_id = (first_id_index[0], first_id_index[1] - 1)
+
+        start_counting_text = None
+
+        if first_id_index == (0, 0):
+            start_counting_text = TOP_LEFT_TEXT
```

Comment:
i think you can refactor this and the following lines out, and give TOP_LEFT_TEXT, right_of_first_id as arguments. besides that they are duplicated minus the conditional above it